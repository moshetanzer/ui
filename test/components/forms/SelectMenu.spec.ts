import { describe, it, expect } from 'vitest'
import { USelectMenu } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('SelectMenu', () => {
  it.each([
    ['basic case', {}],
    ['with custom name', { props: { name: 'SelectMenuName' } }],
    ['width required', { props: { required: true } }],
    ['width icon', { props: { icon: 'i-heroicons-academic-cap' } }],
    ['width searchable', { props: { searchable: true } }],
    ['with options', {
      props: { options: [
        { label: 'Option 1', value: 'option1' },
        { label: 'Option 2', value: 'option2' },
        { label: 'Option 3', value: 'option3', disabled: true }
      ]
    } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof USelectMenu.props>) => {
    const html = await ComponentRender(nameOrHtml, options, USelectMenu)
    expect(html).toMatchSnapshot()
  })
})
