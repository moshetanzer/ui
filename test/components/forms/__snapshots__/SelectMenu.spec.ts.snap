// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectMenu > renders basic case correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <!--v-if-->
  <div id="headlessui-listbox-button-0" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" role="button" class="inline-flex w-full"><button class="relative w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 inline-flex items-center text-left cursor-default rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" type="button">
      <!--v-if--><span class="block truncate"> </span><span class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`SelectMenu > renders width icon correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <!--v-if-->
  <div id="headlessui-listbox-button-3" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" role="button" class="inline-flex w-full"><button class="relative w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 inline-flex items-center text-left cursor-default rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 ps-9 pe-9" type="button"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-academic-cap flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></span><span class="block truncate"> </span><span class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`SelectMenu > renders width required correctly 1`] = `
"<div class="relative" data-headlessui-state=""><input required="" class="absolute inset-0 w-px opacity-0 cursor-default" tabindex="-1" aria-hidden="true">
  <div id="headlessui-listbox-button-2" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" role="button" class="inline-flex w-full"><button class="relative w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 inline-flex items-center text-left cursor-default rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" type="button">
      <!--v-if--><span class="block truncate"> </span><span class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`SelectMenu > renders width searchable correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <!--v-if-->
  <div id="headlessui-combobox-button-4" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" role="button" class="inline-flex w-full"><button class="relative w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 inline-flex items-center text-left cursor-default rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" type="button">
      <!--v-if--><span class="block truncate"> </span><span class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`SelectMenu > renders with custom name correctly 1`] = `
"<input style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;" type="hidden" hidden="" readonly="" name="SelectMenuName">
<div class="relative" data-headlessui-state="">
  <!--v-if-->
  <div id="headlessui-listbox-button-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" role="button" class="inline-flex w-full"><button class="relative w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 inline-flex items-center text-left cursor-default rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" type="button">
      <!--v-if--><span class="block truncate"> </span><span class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
    </button></div>
  <!--v-if-->
</div>"
`;

exports[`SelectMenu > renders with options correctly 1`] = `
"<div class="relative" data-headlessui-state="">
  <!--v-if-->
  <div id="headlessui-listbox-button-5" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" role="button" class="inline-flex w-full"><button class="relative w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0 inline-flex items-center text-left cursor-default rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 shadow-sm bg-white dark:bg-gray-900 text-gray-900 dark:text-white ring-1 ring-inset ring-gray-300 dark:ring-gray-700 focus:ring-2 focus:ring-primary-500 dark:focus:ring-primary-400 pe-9" type="button">
      <!--v-if--><span class="block truncate"> </span><span class="absolute inset-y-0 end-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5" aria-hidden="true"></span></span>
    </button></div>
  <!--v-if-->
</div>"
`;
